src/python/espressomd/observables.py

```diff
@@ -68,6 +69,16 @@ def bin_centers(self):
         shape = list(map(len, edges)) + [len(edges)]
         return np.array(list(itertools.product(*edges))).reshape(shape)
 
+class CylindricalProfileObservable(ProfileObservable):
+    """
+    Base class for observables that work with cylinder coordinates
+    """
+    def __init__(self, **kwargs):
+        if "oid" not in kwargs and "cyl_trafo_params" not in kwargs:
```

Comment:
what is this "oid" check for? can't you do:
``` python
def __init__(self, cyl_trafo_params=CylTrafoParams(), **kwargs):
    super().__init__(**kwargs)
```

Reply:
I think the "oid" is the object identifier for building an object from a checkpoint. Also your example doesn't use the variable `cyl_trafo_params` in the call to the parent class constructor.

Reply:
``` python
def __init__(self, cyl_trafo_params=CylTrafoParams(), **kwargs):
    kwargs['cyl_trafo_params'] = cyl_trafo_params
    super().__init__(**kwargs)
```
?

Reply:
checkpointing does not use constructors

Reply:
My bad, the "oid" stuff was replaced by the "sip" framework, and it's only relevant in special cases like constraints (5a67109596ebd). The highlighted code is a duplication of the "oid" check I added by in the RDF observable, but it's most likely not needed. Also the "sip" documentation is wrong. I'll look into it and open a PR.

Reply:
I will use @KaiSzuttor's suggestion then.
Are there general concerns about how the CylTrafoParams work or the naming and that sort of stuff?
If not I will do docs and tutorials etc

Reply:
Regarding naming, it depends on whether we plan on having other coordinate systems in the future. For example, following this pattern we would have something like `spher_trafo_params` for spherical and `cart_trafo_params` for cartesian (e.g. for a rectangular channel connecting two chambers at an angle).

Or we could name the parameter `coord_system` and give a full name to the coordinate system parameter classes, and throw an error if a `SphericalTransformationParameters` object is passed to a cylindrical observable instead of a `CylindricalTransformationParameters` object. That would be my personal preference.

Reply:
I think that the generic name ```coord_system``` does not directly map to transformation parameters. However, using @jngrad 's idea, I suggest to use ```transform_params``` and have ```CylindricalTransformationParameters```. What do you think?

Reply:
I'm fine with the new names and the suggested change in https://github.com/espressomd/espresso/pull/4114#discussion_r588326876.

Reply:
ok so the observables get the member ``transform_params`` and the CylTrafoParams become ``CylindricalTransformationParameters``.